api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -207,24 +203,56 @@ private Workspace createWorkspaceImpl(Workspace workspace) {
     validateWorkspaceApiModel(workspace);
 
     DbCdrVersion cdrVersion = getLiveCdrVersionId(workspace.getCdrVersionId());
-
+    DbAccessTier accessTier = cdrVersion.getAccessTier();
     DbUser user = userProvider.get();
-    final DbBillingProjectBufferEntry bufferedBillingProject;
-    try {
-      bufferedBillingProject =
-          billingProjectBufferService.assignBillingProject(user, cdrVersion.getAccessTier());
-    } catch (EmptyBufferException e) {
-      throw new TooManyRequestsException(e);
-    }
-    final String billingProject = bufferedBillingProject.getFireCloudProjectName();
 
-    // Note: please keep any initialization logic here in sync with CloneWorkspace().
+    // Note: please keep any initialization logic here in sync with cloneWorkspaceImpl().
+    final String billingProject = claimBillingProject(user, accessTier);
     FirecloudWorkspaceId workspaceId =
         generateFirecloudWorkspaceId(billingProject, workspace.getName());
-    FirecloudWorkspace fcWorkspace = attemptFirecloudWorkspaceCreation(workspaceId);
+    FirecloudWorkspace fcWorkspace =
+        fireCloudService.createWorkspace(
+            workspaceId.getWorkspaceNamespace(),
+            workspaceId.getWorkspaceName(),
+            accessTier.getAuthDomainName());
+
+    DbWorkspace dbWorkspace =
+        createDbWorkspace(workspace, cdrVersion, user, workspaceId, fcWorkspace);
+    try {
```

Comment:
Moved up in file from old line 261.  Old lines 221-261 are now `createDbWorkspace()` because it is common code between create and clone.